runtime/protocol/http-test/jvmAndNative/src/aws/smithy/kotlin/runtime/httptest/TestWithLocalServer.kt

```diff
@@ -33,32 +32,18 @@ public abstract class TestWithLocalServer {
                 attempt++
                 try {
                     server.start()
-                    println("test server listening on: $testHost:$serverPort")
                     break
                 } catch (cause: Throwable) {
                     if (attempt >= 10) throw cause
-                    Thread.sleep(250L * attempt)
+                    delay(250L * attempt)
                 }
             } while (true)
-
-            ensureServerRunning()
```

Comment:
**Question**: Isn't `server.start()` asynchronous? Won't removing `ensureServerRunning()` mean that it's possible to return from this method before the server is _actually_ running?

Reply:
Good point, I refactored this to use ktor's monitoring API. I was having trouble with a KMP solution to `ensureServerRunning` by making a request.